mdn/developer-portal
# 941: Remove redundant static-build code/behaviour

This changeset removes the ability to build (automatically or on demand) a static site and publish it to S3.  We've already moved away from using the S3-hosted site for production when we introduced traditional server-side pagination, so this is a cleanup ticket. No _active_ functional changes are included here, only dropping now-dead code

(Resolves #941)

## Key changes:

- Remove `staticbuild` app from the codebase, which contained all the async tasks and hooks to make automatic build-and-sync run
- Remove `wagtail-bakery` and our custom views for it, which handled the actual building and publishing of the static site
- Add a schema migration to drop the `staticbuild_staticbuild` DB table left behind, orphaned when the `staticbuild` app was removed

## How to test
The code is up on https://developer-portal.dev.mdn.mozit.cloud - have general play around in the CMS and confirm you can publish changes to the content fine. (Because this is a removal PR, checking for things working as expected is all we can do)
